### ISSUE TITLE
[Feature] Restore a minimized window?

### ISSUE THREAD

**Issue by birgersp** [jordansissel/xdotool#430]
Hello,

The `windowminimize` command minimizes a window. But I think a command to restore (unmimize) a minimized window, would be very helpful.

There's `windowmap`, but this doesn't actually restore a minimzed window (?)...

> Map a window. In X11 terminology, mapping a window means making it visible on the
           screen. If no window is given, %1 is the default. See "WINDOW STACK" and "COMMAND
           CHAINING" for more details.

Here's what I had tried:

```
#!/bin/bash

# restore gnome-terminal

PID=$(pgrep -x gnome-terminal-)
if [[ $PID -ne "" ]]
then
    WINDOW_ID=$(xdotool search --pid $PID | tail -1)
    if [[ -n $WINDOW_ID ]]; then
        xdotool windowmap "$WINDOW_ID"
    fi
fi
```

No joy.

It would be nice to see a `windowunminimize` command.

```xdotool windowunminize "$WINDOW_ID"```

**Comment by stormtheory** [comment]
I agree with @birgersp, it would be nice to restore the window once it is minimized. Unless I'm missing something but haven't found a way to restore a window.

**Comment by jordansissel** [comment]
I did a little bit of research and found a few things to note. I checked if `wmctrl` supports this, and I didn't see anything. It's possible there's no direct un-Iconify (unminimize) feature in X11 or the Window Manager protocol EWMH.

Checking on what other projects do, GTK on X11 appears to use [XIconfiyWindow](https://gitlab.gnome.org/GNOME/gtk/-/blob/main/gdk/x11/gdksurface-x11.c#L3272) function to minimize a window, same as xdotool, and also uses [XMapWindow](https://gitlab.gnome.org/GNOME/gtk/-/blob/main/gdk/x11/gdksurface-x11.c#L1530)() when restoring from minimized state.

I'm not exactly sure what the right way is to unminimize something on every window manager. However, I tested on XQuartz on my mac laptop and `windowmap` correctly restores the window after it was minimized.

What windowmanager or desktop environment are you using? I wonder if GNOME or similar handles the "map window" request differently than the XQuartz wm.
